### ISSUE TITLE
Add ability to provision access to connectors using the confluent_role_binding resource

### ISSUE THREAD

**Issue by rerikson** [confluentinc/terraform-provider-confluent#93]
It would be really helpful for our organization's automated provisioning using Terraform if access to Connectors could be provisioned using the confluent_role_binding resource.  I'm assuming this is not yet supported because there is not an example for it in the [documentation](https://github.com/confluentinc/terraform-provider-confluent/blob/master/docs/resources/confluent_role_binding.md) 

I attempted to see if it was supported but not documented using something like the below code snippet, but received HTTP 400 errors in Terraform.

`resource "confluent_role_binding" "connector-example" {
  principal   = "User:${confluent_service_account.test.id}"
  role_name   = "DeveloperWrite"
  crn_pattern = "${confluent_kafka_cluster.basic.rbac_crn}/kafka=${confluent_kafka_cluster.standard.id}/connect=*/connector=*"
}`

Thanks for considering!

**Comment by linouk23** [comment]
👋 @rerikson, thanks for opening an issue!

There could be 2 separate questions:

1. Are you interested in being able to use a service account with `EnvironmentAdmin` / `CloudClusterAdmin` role (instead of having 5-10 ACLs) to provision a connector.

2. Are you interested in having `ConnectorAdmin` role for a service account?

**Comment by rerikson** [comment]
Hey @linouk23 thanks for getting back to me. To answer both questions, My primary use case is actually just to provide access to view a connector in the Confluent Cloud UI for human users in the DeveloperRead or DeveloperWrite roles.

In my organization, those are the only two roles granted to developers, but the developers would like to be able to see the state of their connectors in the Confluent UI.  I'm basically hoping to be able to use Terraform to add additional "Connector" Resource Types to my users like the screenshot below.

![Screen Shot 2022-09-01 at 3 58 22 PM](https://user-images.githubusercontent.com/3589358/188026699-40625221-a8cd-46e5-9415-51cc085ea754.png)

**Comment by linouk23** [comment]
ping @rerikson

**Comment by rerikson** [comment]
Sorry for the delay @linouk23. Our current workaround is to submit a ticket in our ticketing system so that an admin can issue the CLI commands that grant us access to the connector.  The process works, but it gets to be a pain when we're rapidly adding connectors.  

ConnectorReadOnly sounds exactly like what we'd be looking for.

**Comment by linouk23** [comment]
@rerikson thanks for waiting! I talked to one of our teams internally and they said role bindings are supported for connectors and one should look at [this table](https://docs.confluent.io/cloud/current/connectors/managed-connector-rbac.html#connector-rbac-role-mappings) for more details.
![image](https://user-images.githubusercontent.com/5459870/192024330-80bf9fe2-1d14-4f2c-ac1f-9a4d00e8e3b6.png)
It looks like `DeveloperRead` is the role you were looking for.

```
resource "confluent_role_binding" "connector-example-rb" {
  principal   = "User:${confluent_service_account.test.id}"
  role_name   = "DeveloperRead"
  crn_pattern = "${confluent_kafka_cluster.basic.rbac_crn}/connect=${confluent_connector.test.id}"
}
```

For example, you can use Confluent Cloud Console to assign `DeveloperRead` / `DeveloperWrite` roles to Connectors:
![image](https://user-images.githubusercontent.com/5459870/192040799-ef6ab268-fd0c-44d4-803d-3e2bc1367104.png)

**Comment by linouk23** [comment]
Let me know if it helps! We'll update our docs for `confluent_role_binding` resource to include the following example:
```
resource "confluent_role_binding" "connector-example-rb" {
  principal   = "User:${confluent_service_account.test.id}"
  role_name   = "DeveloperRead"
  crn_pattern = "${confluent_kafka_cluster.standard.rbac_crn}/connector=${local.connector_name}"
}

locals {
  connector_name = lookup(confluent_connector.test.config_nonsensitive, "name", "\"name\" attribute is missing")
}
```
in our next release.

**Comment by rerikson** [comment]
That's great to see! I had to check to make sure I didn't somehow miss it and I'm pretty sure the word "Connector" just got added to that table you posted.  I went and took a look using waybackmachine and it's most recent entry didn't have it. https://web.archive.org/web/20220325184735/https://docs.confluent.io/cloud/current/access-management/access-control/cloud-rbac.html

Not that it matters, but I had to know :)
